### PR TITLE
Ability to get more suggestions with and without queries

### PR DESCRIPTION
**Pull request by andrewconner**
Primarily cleanup and optimization. New feature allows clients to paginate through suggestions, fully replacing the current "give me all libraries I can keep to" endpoint. It's cached, so shouldn't be too painful. Curious how well the naïve ranking will do. @leogrim

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/Implicits.scala [hunk 1]
```diff
@@ -40,8 +40,8 @@ final class TryExtensionOps[A](val x: scala.util.Try[A]) extends AnyVal {
     case scala.util.Success(t) => f(t)
     case scala.util.Failure(t) => g(t)
   }
-  def safeOption(implicit airbrake: AirbrakeNotifier): Option[A] = x match {
-    case scala.util.Success(t) => Some(t)
+  def airbrakingOption(implicit airbrake: AirbrakeNotifier): Option[A] = x match {
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/Implicits.scala:43]: Cleared by @ryanpbrewster

File: kifi-backend/modules/common/app/com/keepit/typeahead/Typeahead.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ trait Typeahead[T, E, I, P <: PersonalTypeahead[T, E, I]] extends Logging {
   protected def extractName(info: I): String
 
   protected def buildFilter(id: Id[T], allInfos: Seq[(Id[E], I)]): PrefixFilter[E] = {
-    timing(s"buildFilter($id)") {
+    timing(s"buildFilter($id)", 0.1) {
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/typeahead/Typeahead.scala:50]: I'd like to collect more timing info, so we can see how long these actually take. Could be we can get rid of S3 access.

File: kifi-backend/modules/common/app/com/keepit/common/Implicits.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/common/Implicits.scala:43]: :+1:

File: kifi-backend/modules/shoebox/app/com/keepit/model/LibraryMembershipRepo.scala [hunk 3]
```diff
@@ -305,18 +309,20 @@ class LibraryMembershipRepoImpl @Inject() (
...
-    }
+    relevantSuggestedLibrariesCache.remove(RelevantSuggestedLibrariesKey(libMem.userId))
+    libraryFilterTypeaheadCache.remove(LibraryFilterTypeaheadKey(libMem.userId))
+    countWithLibraryIdByAccessCache.remove(CountWithLibraryIdByAccessKey(libMem.libraryId))
+    countByLibIdAndAccessCache.remove(LibraryMembershipCountByLibIdAndAccessKey(libMem.libraryId, libMem.access))
+    libraryMembershipCountCache.remove(LibraryMembershipCountKey(libMem.userId, libMem.access))
+    if (libMem.canWrite) { librariesWithWriteAccessCache.remove(LibrariesWithWriteAccessUserKey(libMem.userId)) }
+    // ugly! but the library is in an in memory cache so the cost is low
+    val ownerId = libraryRepo.get(libMem.libraryId).ownerId
+    followersCountCache.remove(FollowersCountKey(ownerId))
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/model/LibraryMembershipRepo.scala:320]: None of these actually use `libMem.id`, so might as well clear them even if `id: None`

File: kifi-backend/modules/shoebox/app/com/keepit/typeahead/LibraryTypeahead.scala [hunk 4]
```diff
@@ -28,7 +28,8 @@ object LibraryTypeahead {
 
 @json case class LibraryTypeaheadResult(id: Id[Library], name: String, importance: Int) {
   // importance is a measure to assist in ranking of libraries not recently used by the user. Sum of (lowest is best):
-  //  • -2 active membership
+  //  • -1 active membership
+  //  • -1 user is library owner
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/typeahead/LibraryTypeahead.scala:32]: Fun stat: given org library and library is user created, 83% of keeps in past month from the keeper are to libraries owned by that user.

**Pull request merged**